core/src/main/java/org/hisp/dhis/android/core/attribute/Attribute.java

```diff
@@ -0,0 +1,270 @@
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean mandatory();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean indicatorAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean indicatorGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean userGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean dataElementAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean constantAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean categoryOptionAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean optionSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean sqlViewAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean legendSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean trackedEntityAttributeAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean organisationUnitAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean dataSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean documentAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean validationRuleGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean dataElementGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean sectionAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean trackedEntityTypeAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean userAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean categoryOptionGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean programStageAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean programAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean categoryAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean categoryOptionComboAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean categoryOptionGroupSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean validationRuleAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean programIndicatorAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean organisationUnitGroupAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean dataElementGroupSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean organisationUnitGroupSetAttribute();
+
+    @Nullable
+    @JsonProperty()
+    public abstract Boolean optionAttribute();
+    
+
+    public static Builder builder() {
+        return new $$AutoValue_Attribute.Builder();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.